src/Symfony/Component/Intl/DateFormatter/DateFormat/AmPmUpperCaseTransformer.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Intl\DateFormatter\DateFormat;
+
+/**
+ * Parser and formatter for AM/PM markers format.
+ *
+ * @author Igor Wiedler <igor@wiedler.ch>
+ *
+ * @internal
+ */
+class AmPmUpperCaseTransformer extends Transformer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function format(\DateTime $dateTime, $length)
+    {
+        return $dateTime->format('a');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getReverseMatchingRegExp($length)
+    {
+        return 'am|pm';
```

Comment:
Since even spec uses upper and lower case, and does not differ them, what should actually be done here, is 
'am|pm|AM|PM'